desktop/core/src/desktop/api2.py

```diff
@@ -170,96 +169,53 @@ def get_context_namespaces(request, interface):
-        adb_clusters = []
-
-      if IS_K8S_ONLY.get():
-        sdx_namespaces = []
-      else:
-        sdx_namespaces = SdxApi(request.user).list_namespaces()
-
-      # Adding "fake" namespace for cluster without one
-      sdx_namespaces.extend([_cluster for _cluster in adb_clusters if not _cluster.get('namespaceCrn') or \
-        (IS_K8S_ONLY.get() and 'TERMINAT' not in _cluster['status'])])
-
-      namespaces.extend([{
-          'id': namespace.get('crn', 'None'),
-          'name': namespace.get('namespaceName'),
-          'status': namespace.get('status'),
-          'computes': [_cluster for _cluster in adb_clusters if _cluster.get('namespaceCrn') == namespace.get('crn')]
-        } for namespace in sdx_namespaces if namespace.get('status') == 'CREATED' or IS_K8S_ONLY.get()
-      ])
+  ns_objs = Namespace.objects.filter(dialect=interface)
+  if ns_objs:
+    namespaces = [{
+        'id': ns.id,
+        'name': ns.name,
+        'status': 'CREATED',
+        'computes': [{'id': c['id'], 'type': c['type'], 'name': c['name'], 'dialect': c['dialect'], 'interface': c['interface']} for c in ns.get_computes(request.user)]
+      } for ns in ns_objs
+    ]
+    namespaces = [ns for ns in namespaces if ns['computes']]
+  else:
+    # Currently broken if not sent
+    clusters = list(get_clusters(request.user).values())
+    namespaces.extend([{
+        'id': cluster['id'],
+        'name': cluster['name'],
+        'status': 'CREATED',
+        'computes': [cluster]
+      } for cluster in clusters if cluster.get('type') == 'direct'
+    ])
 
   response[interface] = namespaces
   response['status'] = 0
 
   return JsonResponse(response)
 
-
 @api_error_handler
 def get_context_computes(request, interface):
   '''
   Some clusters like Snowball can have multiple computes for a certain languages (Hive, Impala...).
   '''
   response = {}
-  computes = []
-
-  clusters = list(get_clusters(request.user).values())
-
-  if get_cluster_config(request.user)['has_computes']: # TODO: only based on interface selected?
-    interpreter = get_interpreter(connector_type=interface, user=request.user)
-    if interpreter['dialect'] == 'impala':
-      # dw_clusters = DataWarehouse2Api(request.user).list_clusters()['clusters']
-      dw_clusters = [
-        {'crn': 'c1', 'clusterName': 'c1', 'status': 'created', 'options': {'server_host': 'c1.gethue.com', 'server_port': 10000}},
-        {'crn': 'c2', 'clusterName': 'c2', 'status': 'created', 'options': {'server_host': 'c2.gethue.com', 'server_port': 10000}},
-      ]
-      computes.extend([{
-          'id': cluster.get('crn'),
-          'name': cluster.get('clusterName'),
-          'status': cluster.get('status'),
-          'namespace': cluster.get('namespaceCrn', cluster.get('crn')),
-          'type': interpreter['dialect'],
-          'options': cluster['options'],
-        } for cluster in dw_clusters]
-      )
-  else:
+
+  ns = Namespace.objects.filter(dialect=interface).first()
```

Comment:
Are we choosing first namespace to default to it or are we expecting only one namespace per dialect?

Reply:
We are expecting only one namespace per dialect at the moment. Think of namespace being one per HMS-dialect combination. Since, we have only one hms right now, we will have only one namespace for each dialect. If and when we start supporting multiple HMS, we might need to make some more changes.

Reply:
Nicely explained, thanks!
Should we add this as a comment?